### PR TITLE
Integrate L2Cap channels for Bluetooth.

### PR DESCRIPTION
**Pull request by Plinzen**
Hello, 

I'd like to add a contribution to your library which we developed in my company Continental Automotive Technologies. Starting with Android 10 / iOS 13 L2Cap channels are supported as additional way of communication using BLE. 

This PR adds:
* Support for opening L2Cap channels for listening (server sockets). (listenL2CapChannel/closeL2CapChannel)
* Enable bidirectional communication between connected devices (writeL2CapChannel/readL2CapChannel on BluetoothDevice class)
* Introduced Logging class which could be used from everywhere in the library. 
* Extraction of some marshalling functions into an own util class `MarshallingUtil`, which allow reuse from different sources. 
* Use Swift for newer code (interoperability is given - I don't have ObjC knowledge).

This is a quite big PR, so I am happy to help bringing this feature in. :)

### CODE REVIEW

**Comment by chipweinberger** [comment]
I'm open to adding this. but,  it does not follow FBP conventions.

* All java code should be in a single file
* All iOS code should be in a single file & in obj-c
* all messages should start with Bm and be in the bm_messages.dart file
* no need for "MarshallingUtil" or the logging changes. that should not be in this PR
* `case L2CapMethodNames.CONNECT_TO_L2CAP_CHANNEL` should just be "connectToL2CapChannel"

basically, you're not following FBP conventions at all. 

please open a new PR if you'd like to merge it =)

**Comment by Plinzen** [comment]
Hi, 

thanks for your feedback. Do you have some contribution guidelines available beside the mentioned points? I checked for a `CONTRIBUTING.MD` file, but didn't find one. 
I wanted to use clear separation of concerns and oop techniques within the implementation, but I can change to implement everything into a single file for sure. ObjC might be difficult, I don't have anyone with skill on that topic in my team. Let's see. 

BR

**Comment by chipweinberger** [comment]
no worries

it's normal to follow the existing style when contributing
